src/main/java/org/folio/linked/data/validation/dto/LccnUniquenessValidator.java

```diff
@@ -0,0 +1,83 @@
+package org.folio.linked.data.validation.dto;
+
+import static java.util.Optional.ofNullable;
+
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.linked.data.domain.dto.InstanceField;
+import org.folio.linked.data.domain.dto.LccnField;
+import org.folio.linked.data.domain.dto.LccnRequest;
+import org.folio.linked.data.domain.dto.ResourceRequestDto;
+import org.folio.linked.data.domain.dto.SearchResponseTotalOnly;
+import org.folio.linked.data.exception.RequestProcessingExceptionBuilder;
+import org.folio.linked.data.repo.FolioMetadataRepository;
+import org.folio.linked.data.service.search.InstanceSearchService;
+import org.folio.linked.data.validation.LccnUniqueConstraint;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Log4j2
+public class LccnUniquenessValidator extends LccnValidator<LccnUniqueConstraint, ResourceRequestDto> {
+
+  private final InstanceSearchService instanceSearchService;
+  private final FolioMetadataRepository folioMetadataRepository;
+  private final RequestProcessingExceptionBuilder exceptionBuilder;
+
+  @Override
+  public boolean isValid(ResourceRequestDto resourceRequestDto, ConstraintValidatorContext constraintValidatorContext) {
+    var resource = resourceRequestDto.getResource();
+    if (resource instanceof InstanceField instance && hasCurrentLccn(instance)) {
+      var inventoryId = findInventoryId(resourceRequestDto);
+      return isUnique(getLccnValues(instance), inventoryId);
+    }
+    return true;
+  }
+
+  private String findInventoryId(ResourceRequestDto resourceRequestDto) {
+    return ofNullable(resourceRequestDto.getId())
+      .flatMap(folioMetadataRepository::findInventoryIdById)
+      .map(FolioMetadataRepository.InventoryIdOnly::getInventoryId)
+      .orElse(null);
+  }
+
+  private boolean isUnique(List<String> lccn, String inventoryId) {
+    return ofNullable(findInstanceWithLccn(lccn, inventoryId))
+      .map(SearchResponseTotalOnly::getTotalRecords)
+      .map(count -> count == 0)
+      .orElse(false);
+  }
+
+  private SearchResponseTotalOnly findInstanceWithLccn(List<String> lccn, String inventoryId) {
+    try {
+      return instanceSearchService.searchByLccnExcludingId(lccn, inventoryId);
+    } catch (Exception e) {
```

Comment:
we already catch and wrap feign exceptions inside of instanceSearchService, so I don't think we need to catch anything here. Do you see a reason?

Reply:
it's needed to build clear message for user that validation failed due to search service.